### PR TITLE
Fix issue Provider can't make offers if candidate has other applications with conditions_not_met

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

If candidate accepts an offer from application choice (e.g 1) and application choice 1 endup in conditions not met

And the candidate applies to another application choice (e.g 2) they should be able to receive offers from any other choice

## Impact

There are 22 candidates with conditions not met and they can't progress with their other applications.

## How many times it happen?

41 times in this month 
https://dfe-teacher-services.sentry.io/issues/4385529849/?alert_rule_id=4434951&alert_type=issue&environment=production&notification_uuid=1277a5dd-1927-4fab-8e62-c9f4a8b3ad3b&project=1765973&referrer=slack

## Example in QA

https://qa.apply-for-teacher-training.service.gov.uk/support/applications/36807

## Guidance to review

1. Have a candidate with two applications: 1 in conditions not met and other awaiting provider decision
2. Make an offer to the awaiting provider decision and the offer should reach the candidate

## Link to Trello card

Support ticket:
https://trello.com/c/21xhaeAD/2725-issue-around-making-offer-to-candidate

Candidate ticket:
https://trello.com/c/4wVNbRwy/1359-fix-provider-not-being-able-to-make-an-offer

### CODE REVIEW

**Comment by avinhurry** [comment]
[Fix provider not being able to make an offer](https://trello.com/c/4wVNbRwy/1359-fix-provider-not-being-able-to-make-an-offer)

**Comment by avinhurry** [comment]
[Issue around making offer to candidate](https://trello.com/c/21xhaeAD/2725-issue-around-making-offer-to-candidate)

File: app/services/offer_validations.rb [hunk 1]
```diff
@@ -99,7 +99,9 @@ def reference_condition
   end
 
   def any_accepted_offers?
-    (application_choice.self_and_siblings - [application_choice]).map(&:status).map(&:to_sym).intersect?(ApplicationStateChange::ACCEPTED_STATES)
+    (application_choice.self_and_siblings - [application_choice])
+      .map(&:status).map(&:to_sym)
+      .intersect?(ApplicationStateChange::ACCEPTED_STATES - [:conditions_not_met])
```
Comment (inulty-dfe) [inline app/services/offer_validations.rb:104]: This method doesn't make sense anymore. `conditions_not_met` is and accepted offer state. The method name says **any**_accepted_offers?

Maybe we should create a method somewhere `can_receive_other_offers?` or something along those lines?

**Review by inulty-dfe — CHANGES_REQUESTED** [review]
Method name doesn't make sense anymore with these changes

File: app/services/offer_validations.rb (hunk 1 above)
Comment (avinhurry) [reply app/services/offer_validations.rb:104]: Updated here: https://github.com/DFE-Digital/apply-for-teacher-training/pull/9141/commits/f29c27fcba8b0e2f4d1f663a802ee06e0298feb4

File: spec/services/offer_validations_spec.rb [hunk 2]
```diff
@@ -138,6 +138,16 @@
         end
       end
 
+      context 'when a provider attemps to make an offer with other application as conditions not met' do
```
Comment (inulty-dfe) [inline spec/services/offer_validations_spec.rb:141]:
```suggestion
      context 'when a provider attempts to make an offer when conditions not met sibling exists' do
```

File: spec/services/offer_validations_spec.rb [hunk 3]
```diff
@@ -138,6 +138,16 @@
...
+        let!(:application_form) { create(:application_form, application_choices: [application_choice, other_application_choice]) }
+        let(:application_choice) { build(:application_choice, :offered, current_course_option: course_option) }
+        let!(:other_application_choice) { build(:application_choice, :conditions_not_met) }
+
+        it 'be valid' do
```
Comment (inulty-dfe) [inline spec/services/offer_validations_spec.rb:146]:
```suggestion
        it 'is valid' do
```

**Review by inulty-dfe — APPROVED** [review]
Couple suggestions on the spec descriptions

File: spec/services/offer_validations_spec.rb (hunk 2 above)
Comment (avinhurry) [reply spec/services/offer_validations_spec.rb]: https://github.com/DFE-Digital/apply-for-teacher-training/pull/9141/commits/f78a6bf173dff8eb57aefcd472371c6ad5589a45

File: spec/services/offer_validations_spec.rb (hunk 3 above)
Comment (avinhurry) [reply spec/services/offer_validations_spec.rb]: https://github.com/DFE-Digital/apply-for-teacher-training/pull/9141/commits/f78a6bf173dff8eb57aefcd472371c6ad5589a45

**Pull request merged**